src/modules/musig/keyagg.h

```diff
@@ -0,0 +1,40 @@
+/***********************************************************************
+ * Copyright (c) 2021 Jonas Nick                                       *
+ * Distributed under the MIT software license, see the accompanying    *
+ * file COPYING or https://www.opensource.org/licenses/mit-license.php.*
+ ***********************************************************************/
+
+#ifndef SECP256K1_MODULE_MUSIG_KEYAGG_H
+#define SECP256K1_MODULE_MUSIG_KEYAGG_H
+
+#include "../../../include/secp256k1.h"
+#include "../../../include/secp256k1_musig.h"
+
+#include "../../field.h"
+#include "../../group.h"
+#include "../../scalar.h"
+
+typedef struct {
+    secp256k1_ge pk;
+    /* If there is no "second" public key, second_pk is set to the point at
+     * infinity */
+    secp256k1_ge second_pk;
+    unsigned char pk_hash[32];
+    /* tweak is identical to value tacc[v] in the specification. */
+    secp256k1_scalar tweak;
+    /* parity_acc corresponds to gacc[v] in the spec. If gacc[v] is -1,
+     * parity_acc is 1. Otherwise, parity_acc is 0. */
+    int parity_acc;
+} secp256k1_keyagg_cache_internal;
+
+/* musig_ge_to_bytes_ext and musig_ge_from_bytes_ext are identical to ge_save and ge_load
+ * except that they allow saving and loading the point at infinity */
+static void secp256k1_musig_ge_to_bytes_ext(unsigned char *data, secp256k1_ge *ge);
+
+static void secp256k1_musig_ge_from_bytes_ext(secp256k1_ge *ge, const unsigned char *data);
```

Comment:
I think these should be in `group.h`, even though they'll be used only by the musig module. Infinity can be helpful in other contexts, and conceptually it's a group function.

Anyway, the comment needs to be updated `ge_save` and `ge_load` have been renamed.

Reply:
Agree, done.